### PR TITLE
Added isocontour shader

### PR DESCRIPTION
**Pull request by sochowski**
- Created `isocontour.frag.glsl` that renders points within the range of `iso_tolerance` from any `iso_layers[i]`, where `i` is in the range of [0, `iso_num_layers`)
- Added slider to adjust `iso_tolerance`
- Added buttons to add and remove isocontour layers, as well as the ability to input up to 32 distinct layers

### CODE REVIEW

File: yt_idv/scene_components/base_component.py [hunk 1]
```diff
@@ -83,17 +83,23 @@ def render_gui(self, imgui, renderer, scene):
...
                 self.iso_layers.append(0.0)
-        if imgui.button("Remove Layer"):
-            if len(self.iso_layers) > 0:
-                changed = True
-                self.iso_layers.pop()
-        for i in range(len(self.iso_layers)):
-            changed, self.iso_layers[i] = imgui.input_float(
+        imgui.columns(2, "iso_layers_cols", False)
+        i = 0
+        while i < len(self.iso_layers):
```
Comment (matthewturk) [inline yt_idv/scene_components/base_component.py:88]: I'm trying to figure out how to manage this without a `while` loop, but I can't quite come up with something.  So I suppose, this is OK!  :)

File: yt_idv/scene_components/base_component.py [hunk 2]
```diff
@@ -260,3 +295,6 @@ def run_program(self, scene):
 
     def draw(self, scene, program):
         raise NotImplementedError
+
+    def _get_sanitized_iso_layers(self):
```
Comment (chrishavlin) [inline yt_idv/scene_components/base_component.py:299]:
```suggestion
    def _get_sanitized_iso_layers(self):
        # child scene_component classes override this to normalize layer values
```

File: yt_idv/scene_components/base_component.py (hunk 1 above)
Comment (chrishavlin) [reply yt_idv/scene_components/base_component.py:88]: I also think this is good for now! But I also agree that there could a way to simplify this... maybe separating the initial table construction from row removal when a button is pushed? Would result in more code but might be conceptually simpler. But ya, I say leave this as is for now.

**Review by chrishavlin — APPROVED** [review]
This version looks good! One small code suggestion to add a comment, but good to merge after that.

**Pull request merged**